packages/ui/src/lib/button/CloseButton.svelte

```diff
@@ -2,26 +2,21 @@
 import { faTimes } from '@fortawesome/free-solid-svg-icons';
 import { createEventDispatcher } from 'svelte';
 import Fa from 'svelte-fa';
-import { router } from 'tinro';
 
-export let href: string | undefined = undefined;
+export let href: string | undefined = 'undefined';
```

Comment:
I wonder why we are keeping the href here. Before there was a different behavior based on the value href had. If it was undefined a click was dispatched, if it was defined it worked like a link. 
In the new way we always trigger the click. Can we remove this href?

Reply:
ah it's a leftover from a test, it should not be 'undefined'

Reply:
so now I see it's a suggestion

the idea is that if you expect an href you'll get a string, else you receive undefined

Reply:
you always get a click event, then it's either containing a value or undefined

Reply:
I kept the href so when we get the click event we receive the value, the won't need to keep it on the parent component as an attribute

Reply:
but we could replace `href` by `clickDetail` or 'detail`

Reply:
it looks to me too complex tbh, we could just register a callback which gets called on the click of the close button.
If you want to keep the current logic, yes maybe renaming would reduce the confusion imo

Reply:
> it looks to me too complex tbh, we could just create the close button component with a callback that gets called on the click.

it's what it does, it call the optional on:click method

but I can remove the field

waiting for other reviews to decide

Reply:
I was referring to a simplification like the following (so we remove the href, dispatch and click() from the CloseButton) 
```
export let onClose: () => {};

on:click="{onClose}"
```

then from the caller
```
<CloseButton
   onClose=() => { .... }
```

Reply:
I changed the implementation to match Link implementation